docs/processes/fuzzer_targets.md

```diff
@@ -0,0 +1,119 @@
+======================================
+
+This documents describes how to develop a [fuzzer](https://opensource.google.com/projects/oss-fuzz) target for an ICU API and its integration into the ICU build process.
+
+### Directory and naming conventions
+
+Fuzzer targets are exclusively in directory [```source/test/fuzzer/```](https://github.com/unicode-org/icu/tree/master/icu4c/source/test/fuzzer) and end with ```_fuzzer.cpp```. Only files with such ending are recognized and executed as fuzzer targets by the OSS-Fuzz system.
+
+### General structure of a fuzzer target
+
+As a minimum, a fuzzer target contains the function
+
+```
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size){
+  ...
+}
+```
+
+This function is expected and invoked by the fuzzer system. The ```data``` parameter contains the fuzzer-controled data of size ```size``` bytes. Part or all of this data is then passed into the ICU API under test.
+
+Fuzzer target ```collator_rulebased_fuzzer.cpp``` illustrates the basic elements.
+
+```
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include <cstring>
+
+#include "fuzzer_utils.h"
+#include "unicode/coll.h"
+#include "unicode/localpointer.h"
+#include "unicode/locid.h"
+#include "unicode/tblcoll.h"
+
+IcuEnvironment* env = new IcuEnvironment();
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  UErrorCode status = U_ZERO_ERROR;
+
+  size_t unistr_size = size/2;
+  std::unique_ptr<char16_t[]> fuzzbuff(new char16_t[unistr_size]);
+  std::memcpy(fuzzbuff.get(), data, unistr_size * 2);
+  icu::UnicodeString fuzzstr(false, fuzzbuff.get(), unistr_size);
+
+  icu::LocalPointer<icu::RuleBasedCollator> col1(
+      new icu::RuleBasedCollator(fuzzstr, status));
+
+  return 0;
+}
+```
+
+The ICU API under test is the ```RuleBasedCollator(const UnicodeString &rules, UErrorCode &status)``` constructor. The code interprets the fuzzer data as UnicodeString and passes it to the constructor. And that is all. Specific error handling or return value verification is not required because the fuzzer will detect all memory issues by means of memory/address sanitizer findings.
+
+### Makefile.in changes
+
+ICU fuzzer targets are build and executed by the OSS-Fuzz project. On side of ICU they are compiled to assure that the code is syntactically correct and, as a sanity check, executed in the most basic manner, i.e. with minimal testdata and without ASAN or MSAN analysis.
+
+Add the new fuzzer target to the list of targets in the ```FUZZER_TARGETS``` variable in [```Makefile.in```](https://github.com/unicode-org/icu/blob/master/icu4c/source/test/fuzzer/Makefile.in). The new fuzzer target will then be built and executed as part of a normal ICU4C unit test run. Note that each fuzzer target becomes executable of its own. As such it is linked with the code in ```fuzzer_driver.cpp```, which contains the ```main()``` function.
```

Comment:
Do you know if there is a way to have a shorter link, such as one that is relative to the root of the project?

Reply:
Looked around, unfortunately didn't find. The other markdown docs also have full URL.